app/helpers/articles_helper.rb

```diff
@@ -41,11 +41,12 @@ def should_show_updated_on?(article)
   end
 
   def should_show_crossposted_on?(article)
-    article.crossposted_at &&
+    article.canonical_url ||
```

Comment:
I didn't see a reason about making this more complicated than a simple "if we have a canonical URL, show it". The logic for RSS feeds can probably be simplified too (e.g. if the article isn't published, the user may still want to see the "Originally published at" in their draft), but that's for another day.